### PR TITLE
[JSDK-25] Added public method to retrieve a Currency by its currency code 

### PR DESCRIPTION
**Pull request by acostros**
Added public method to GET a Currency by its currency code

### CODE REVIEW

File: src/com/mambu/apisdk/services/OrganizationService.java [hunk 1]
```diff
@@ -134,6 +137,29 @@ public Currency getCurrency() throws MambuApiException {
 	}
 
 	/**
+	 * Requests a currency for a supplied currency code.
+	 * 
+	 * @param currencyCode
+	 *            the currency code for the currency to be searched in Mambu.
+	 *
```
Comment (mdanilkis) [inline src/com/mambu/apisdk/services/OrganizationService.java]: Can add the standard "Must not be null" thingy to the param

File: src/com/mambu/apisdk/services/OrganizationService.java [hunk 2]
```diff
@@ -134,6 +137,29 @@ public Currency getCurrency() throws MambuApiException {
 	}
 
 	/**
+	 * Requests a currency for a supplied currency code.
+	 * 
+	 * @param currencyCode
+	 *            the currency code for the currency to be searched in Mambu.
+	 * 
+	 * @return Currency having as currency code the currency code passed as parameter to this method call, or
+	 *         INVALID_CURRENCY_CODE error code if there is no currency having the code supplied to this method call.
```
Comment (mdanilkis) [inline src/com/mambu/apisdk/services/OrganizationService.java]: It's nice to have the description of the error case. Can possibly clarify that it is the exception that is thrown in the error scenario (when the error code is returned by Mambu server)

File: src/com/mambu/apisdk/services/OrganizationService.java [hunk 3]
```diff
@@ -134,6 +137,29 @@ public Currency getCurrency() throws MambuApiException {
...
+		// Example: GET api/currencies/{currencyCode}
+
+		if (currencyCode == null) {
+			throw new IllegalArgumentException("The currency code can`t be null!");
+		}
+
+		return serviceExecutor.execute(getCurrencyByCode, currencyCode);
+	}
+
+	/**
```
Comment (mdanilkis) [inline src/com/mambu/apisdk/services/OrganizationService.java]: Nice!

File: src/com/mambu/apisdk/services/OrganizationService.java (hunk 1 above)
Comment (acostros) [inline src/com/mambu/apisdk/services/OrganizationService.java]: yep, sorry just missed it.

File: src/com/mambu/apisdk/services/OrganizationService.java (hunk 2 above)
Comment (acostros) [inline src/com/mambu/apisdk/services/OrganizationService.java]: Yes, just added it. Thanks!

File: src/com/mambu/apisdk/services/OrganizationService.java [hunk 4]
```diff
@@ -134,6 +137,30 @@ public Currency getCurrency() throws MambuApiException {
 	}
 
 	/**
+	 * Requests a currency for a supplied currency code.
+	 * 
+	 * @param currencyCode
+	 *            the currency code for the currency to be searched in Mambu. Must not be null.
+	 * 
+	 * @return Currency having as currency code the currency code passed as parameter to this method call. In case a
+	 *         wrong currency code (there is no currency having currency code having this code) is passed to this method
```
Comment (florinparascan) [inline src/com/mambu/apisdk/services/OrganizationService.java]: Maybe we could replace "there is no currency having currency code having this code" with something like just "there is no currency having this code".

File: src/demo/DemoTestOrganizationService.java [hunk 5]
```diff
@@ -207,14 +216,15 @@ public static void testGetCentresByPage() throws MambuApiException {
...
+		System.out.println(methodName = "\nIn " + methodName);
 		OrganizationService organizationService = MambuAPIFactory.getOrganizationService();
 
 		String branchId = BRANCH_ID;
 		String offset = "0";
 		String limit = "500";
-		System.out.println("\nIn testGetCentresByBranch" + "  BranchID=" + branchId + "  Offset=" + offset + "  Limit="
-				+ limit);
+		System.out.println(
+				"\nIn testGetCentresByBranch" + "  BranchID=" + branchId + "  Offset=" + offset + "  Limit=" + limit);
```
Comment (florinparascan) [inline src/demo/DemoTestOrganizationService.java]: It's very nice that you used the new logging approach to all of the tests. I think we can use it also here. What do you think?

File: src/com/mambu/apisdk/services/OrganizationService.java (hunk 4 above)
Comment (acostros) [inline src/com/mambu/apisdk/services/OrganizationService.java]: yep more concise, thanks!

File: src/demo/DemoTestOrganizationService.java (hunk 5 above)
Comment (acostros) [inline src/demo/DemoTestOrganizationService.java]: good idea, just changed it. Thanks!

**Pull request merged**